install-yarn.js

```diff
@@ -6,8 +6,14 @@ var yarnModule
   , node = process.argv[0]
   ;
 
-if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/node_modules\/yarn\/bin\/yarn\.js$/)) {
-  yarnModule = path.resolve(process.env['npm_execpath'], '..', '..', 'lib', 'cli');
+if (process.env['npm_execpath']) {
+  if (process.platform === 'win32') {
+    process.env['npm_execpath'] = process.env['npm_execpath'].replace(/\\/g, '/');
+  }
+
+  if (process.env['npm_execpath'].match(/node_modules\/yarn\/bin\/yarn\.js$/)) {
```

Comment:
Same goes here, we can rely on node to do the groundwork:
```suggestion
    var expectedYarnPath = path.join('bin', 'yarn.js')
    if (process.env['npm_execpath'].substr(-1 * expectedYarnPath.length) === expectedYarnPath) {
```

Reply:
Latest approach uses path.join to build a path to both NPM and Yarn binaries, then compares to execpath.